source/lac/lapack_full_matrix.cc

```diff
@@ -1956,6 +1957,102 @@ LAPACKFullMatrix<number>::compute_eigenvalues(const bool right, const bool left)
 }
 
 
+
+namespace
+{
+  template <typename RealNumber>
+  unsigned int
+  fill_complex_conjugate_eigenvalue_pair(
+    const unsigned int                    index,
+    const std::vector<RealNumber> &       vr,
+    const std::complex<RealNumber> &      eigenvalue,
+    FullMatrix<std::complex<RealNumber>> &result)
+  {
+    const std::size_t n = result.n();
+    if (eigenvalue.imag() != 0.)
+      {
+        // for real-valued matrices, eigenvalues with a non-zero imaginary
+        // part come in pairs. In LAPACK, a compact storage scheme is applied
+        // that stores the real and imaginary part of eigenvectors only once,
+        // putting the real parts in one column and the imaginary part in the
+        // next. Here, we do the unpacking into the usual complex values.
+        for (std::size_t j = 0; j < n; ++j)
+          {
+            result(j, index).real(vr[index * n + j]);
+            result(j, index + 1).real(vr[index * n + j]);
+            result(j, index).imag(vr[(index + 1) * n + j]);
+            result(j, index + 1).imag(-vr[(index + 1) * n + j]);
+          }
+
+        // we filled two columns, so return that
+        return index + 2;
+      }
+    else
+      {
+        for (unsigned int j = 0; j < n; ++j)
+          result(j, index).real(vr[index * n + j]);
+
+        // real-valued case, we only filled one column
+        return index + 1;
+      }
+  }
+
+  template <typename ComplexNumber>
+  unsigned int
+  fill_complex_conjugate_eigenvalue_pair(const unsigned int index,
+                                         const std::vector<ComplexNumber> &vr,
+                                         const ComplexNumber &,
+                                         FullMatrix<ComplexNumber> &result)
+  {
+    const std::size_t n = result.n();
+    for (unsigned int j = 0; j < n; ++j)
+      result(j, index) = vr[index * n + j];
+
+    // complex-valued case always only fills a single column
+    return index + 1;
+  }
```

Comment:
I was substantially confused when I saw this function used below because it's unclear what the function returns in that context. Could I entice you to document what the returned object indicates? Or maybe name the function `..._and_increment_index()` and make `index` a writable reference?

Reply:
I switched to the suggested version of using `index` as a writable reference. To make the name still fit on one line, I opted for `unpack_lapack_eigenvector_and_increment_index`. I also added a better explanation in that place.